File: lib/domain/agricultural_task/dtos/task_schedule_replace_item.rb
# frozen_string_literal: true

module Domain
  module AgriculturalTask
    module Dtos
      # TaskSchedule 一括置換時に Gateway が永続化する 1 行分（Interactor が組み立てる）。
      class TaskScheduleReplaceItem
        attr_reader :task_type, :agricultural_task_id, :name, :description,
                    :stage_name, :stage_order, :gdd_trigger, :gdd_tolerance,
                    :scheduled_date, :priority, :source, :status,
                    :weather_dependency, :time_per_sqm, :amount, :amount_unit

        def initialize(
          task_type:,
          agricultural_task_id:,
          name:,
          description:,
          stage_name:,
          stage_order:,
          gdd_trigger:,
          gdd_tolerance:,
          scheduled_date:,
          priority:,
          source:,
          status:,
          weather_dependency:,
          time_per_sqm:,
          amount:,
          amount_unit:
        )
          @task_type = task_type
          @agricultural_task_id = agricultural_task_id
          @name = name
          @description = description
          @stage_name = stage_name
          @stage_order = stage_order
          @gdd_trigger = gdd_trigger
          @gdd_tolerance = gdd_tolerance
          @scheduled_date = scheduled_date
          @priority = priority
          @source = source
          @status = status
          @weather_dependency = weather_dependency
          @time_per_sqm = time_per_sqm
          @amount = amount
          @amount_unit = amount_unit
        end
      end
    end
  end
end

File: crates/agrr-domain/src/agricultural_task/dtos/task_schedule_replace_item.rs
use rust_decimal::Decimal;
use time::Date;

/// Item attributes passed to task schedule gateway replace.
#[derive(Debug, Clone)]
pub struct TaskScheduleReplaceItem {
    pub task_type: String,
    pub agricultural_task_id: Option<i64>,
    pub name: String,
    pub description: Option<String>,
    pub stage_name: Option<String>,
    pub stage_order: Option<i32>,
    pub gdd_trigger: Decimal,
    pub gdd_tolerance: Option<Decimal>,
    pub scheduled_date: Date,
    pub priority: Option<i32>,
    pub source: Option<String>,
    pub status: String,
    pub weather_dependency: Option<String>,
    pub time_per_sqm: Option<Decimal>,
    pub amount: Option<Decimal>,
    pub amount_unit: Option<String>,
}
